docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.daysInMonth](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.daysInYear](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.leapYear](
+		self: Temporal.Date
+	) : number;
+}
+```
+
+An instance of `MyCalendar` is expected to have stateless behavior; i.e., calling a method with the same arguments should return the same result each time.
+
+## Default Calendar
+
+An open question is what the behavior should be if the programmer does not specify a calendar, or if we should require the programmer to always specify a calendar.  Four choices are on the table:
+
+1. Default to full ISO (Gregorian) calendar.
+2. Require the user to explicitly specify the calendar.
+3. Default to a partial ISO calendar (explained below).
+4. Default to `Intl.defaultCalendar` (a new symbol), or ISO if that field doesn't exist.
+
+### Partial ISO Calendar (Option 3)
+
+A partial ISO calendar would be one implemented as follows:
+
+```javascript
+const PartialIsoCalendar = {
+	[Temporal.Calendar.toISO] = (self) => {
+		return self;
+	},
+
+	[Temporal.Calendar.fromISO] = (isoDate) => {
+		return isoDate;
+	},
+
+	[Temporal.Calendar.id] = "iso",
+
+	// ALL OTHER METHODS:
+	[Temporal.Calendar.plus] = () => {
+		throw new TypeError("Unsupported operation: full calendar required");
+	}
+	// Same for [Temporal.Calendar.minus], etc.
+}
+```
+
+It would in effect render default Temporal.Date (and Temporal.DateTime) with fewer operations until you specify a calendar.  The following methods/getters would throw:
+
+- .dayOfWeek
+- .weekOfYear
+- .daysInMonth
+- .daysInYear
+- .leapYear
+- .plus() -- *might* be OK if the programmer requests only time units
+- .minus() -- *might* be OK if the programmer requests only time units
+- .difference() -- *might* be OK if the programmer requests only time units
+- .getYearMonth()
+- .getMonthDay()
+
+The following methods/getters would still work:
+
+- .with()
+- .withTime()
+- .toString()
+- .toLocaleString()
+- .compare()
+
+Although small, this set of operations still covers many of the recipes in the Temporal Cookbook.
+
+To enable the extended set of operations, the user would just use `.withCalendar()`:
+
+```javascript
+// Force the Gregorian calendar:
+Temporal.Date.from("2019-12-06").withCalendar("gregory").weekOfYear;
+
+// Use a calendar from another source:
+Temporal.Date.from("2019-12-06").withCalendar(Intl.defaultCalendar).weekOfYear;
+Temporal.Date.from("2019-12-06").withCalendar(request.calendar).weekOfYear;
+```
+
+The Gregorian calendar, ID "gregory" ([why not "gregorian"?](https://github.com/tc39/ecma402/issues/212)), could be used to indicate the full ISO calendar.  So, "iso" would mean the partial version, and "gregory" would mean the full version.  Alternatively, "iso8601", the identifier defined by CLDR as "Gregorian calendar using the ISO 8601 calendar week rules", could be the identifier for the full ISO calendar.
```

Comment:
This would be very confusing, since there is a gregorian calendar with differing time rules (ethiopian) as well.

The calendar with ISO date, time and week rules should be called iso. Having iso be some crippled partial version is not a good idea.

Reply:
It sounds like this question is one of naming.  If you want `"iso"` to mean the full ISO calendar, how about `null` (or `undefined` / `""` ?) for the crippled partial version and `"iso"` for the full version?

Reply:
Iff we have a crippled iso calendar, this is a question of naming.

I’d rather not have a crippled calendar.

Reply:
The other thing was that I would expect Temporal.Date.prototype.toISO() to return an object with the default calendar, which in Option 3 would be partial ISO.  So basically partial ISO is the default entrypoint, and then all other calendars, full ISO included, are upgrades that the user can request, and to convert from one upgrade to another upgrade, you pass back through the default partial ISO calendar.

Reply:
I rewrote the paragraph to read:

> The calendar IDs are less clear.  If the partial ISO calendar used ID `"iso"`, then what would the full ISO calendar use?  ID "gregory" ([why not "gregorian"?](https://github.com/tc39/ecma402/issues/212)) is misleading because there are Gregorian calendars that do not all agree on the same rules for things like weeks of the year.  One solution could be to use a nullish ID like `null` or `""` for the partial ISO calendar and `"iso"` for the full ISO calendar.  Alternatively, "iso8601", the identifier defined by CLDR as "Gregorian calendar using the ISO 8601 calendar week rules", could be the identifier for the full ISO calendar.

Resolving this thread.

Reply:
https://github.com/tc39/proposal-temporal/pull/287#discussion_r355206573

That’s only true if you consider option 3 an actual option. I still consider it the worst of all the options and only on the list because it would at least be feasible 😀